rust/progress_reporting/src/data_progress.rs

```diff
@@ -143,6 +143,11 @@ impl DataProgressReporter {
 
     /// Does the actual printing
     fn print(&self, is_final: bool) -> std::result::Result<(), std::io::Error> {
+        // Put a minimum width for the line we print.  This is based on git's messages,
+        // which can be longer than ours, causing weird inteleaving when the user does
+        // stuff to the input (hits a new line, etc.)  So, print a minimum buffer of spaces.
+        const PRINT_LINE_MIN_WIDTH: usize = 74;
```

Comment:
Wondering why choosing 74

Reply:
Measured the length of the git progress strings that the original bug report showed.